.github/workflows/build.yml

```diff
@@ -0,0 +1,152 @@
+name: "Build"
+
+on:
+  push:
+  pull_request:
+
+jobs:
+
+  test:
+    name: Maven Test & Package
+    runs-on: ubuntu-latest
+
+    env:
+      JAVA_DISTRIBUTION: microsoft
+      JAVA_VERSION: 17
+
+    services:
+      postgres:
+        image: postgres:13.6
+        ports:
+          - 5432:5432
+        env:
+           POSTGRES_DB: train-handler-test
+           POSTGRES_USER: train-handler
+           POSTGRES_PASSWORD: password
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Fetch all history for all tags and branches
+        run: git fetch --prune --unshallow
+
+      # (1) -> Prepare cache
+      - name: Cache ~/.m2
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-${{ env.JAVA_VERSION }}-maven-${{ hashFiles('**/pom.xml') }}
+
+      # (2) -> Prepare Java
+      - name: Setup Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: ${{ env.JAVA_DISTRIBUTION }}
+          java-version: ${{ env.JAVA_VERSION }}
```

Comment:
Cache can be simplified with newer versions of setup-java I believe. See https://github.com/actions/setup-java/blob/main/README.md#caching-maven-dependencies for more info.